Move bash completion scripts for geopmread & write to service

avilcheslopez opened the pull request:
- Fixes #2872

avilcheslopez committed:
Move bash completion scripts for geopmread & write to service

Signed-off-by: Alejandro Vilches <alejandro.vilches@intel.com>

cmcantalupo committed:
Echo config log if configure fails in github CI

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

avilcheslopez committed:
Fix license test, bash-completion dependency

- Add service/shell_completion/README.md to exempt manifest
- On CentOS 9, bash-completion package provides bash completion
  feature (not bash-completion-devel)

Signed-off-by: Alejandro Vilches <alejandro.vilches@intel.com>

dannosliwcd commented on service/Makefile.am:
```diff
@@ -437,6 +437,7 @@ include test/Makefile.mk
 if !ENABLE_SANITIZERS
 include geopmdpy_test/Makefile.mk
 include integration/Makefile.mk
+include shell_completion/Makefile.mk
```
Why do shell completion scripts depend on the sanitizer config?

cmcantalupo commented on service/Makefile.am:
```diff
@@ -437,6 +437,7 @@ include test/Makefile.mk
 if !ENABLE_SANITIZERS
 include geopmdpy_test/Makefile.mk
 include integration/Makefile.mk
+include shell_completion/Makefile.mk
 endif
 
 include fuzz_test/Makefile.mk
```
```suggestion
endif

include shell_completion/Makefile.mk

include fuzz_test/Makefile.mk
```

cmcantalupo committed:
Include shell completion makefile when sanitizers are enabled.

- Logic to exclude makefiles is to avoid running the sanitizers against the test code.
- Since there is no C or C++ code in the bash_completion makefile, it is safe to include it.

cmcantalupo commented on .github/workflows/build.yml:
```diff
@@ -137,15 +137,15 @@ jobs:
     - name: configure service dir
       working-directory: service
-      run: ./autogen.sh && ./configure
+      run: ./autogen.sh && ./configure || cat config.log && false
```
```suggestion
      run: ./autogen.sh && ./configure || (cat config.log && false)
```

cmcantalupo commented on .github/workflows/build.yml:
```diff
@@ -77,13 +77,13 @@ jobs:
     - name: configure service dir
       working-directory: service
-      run: ./autogen.sh && ./configure --${{ matrix.debug-flag }} --${{ matrix.service-asan-flag }}
+      run: ./autogen.sh && ./configure --${{ matrix.debug-flag }} --${{ matrix.service-asan-flag }}  || (cat config.log && false)
```
```suggestion
      run: ./autogen.sh && ./configure --${{ matrix.debug-flag }} --${{ matrix.service-asan-flag }} || (cat config.log && false)
```

cmcantalupo committed:
Clean up change to emit config log when configure fails

cmcantalupo committed:
Update gcc, clang and libomp to latest packages available on jammy

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>
